// @angular/common/src/directives/ng_if.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
import * as import0 from '@angular/common/src/directives/ng_if';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
export var Wrapper_NgIf = (function () {
    function Wrapper_NgIf(p0, p1) {
        this.changed = false;
        this.context = new import0.NgIf(p0, p1);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_NgIf.prototype.check_ngIf = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.ngIf = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgIf.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    Wrapper_NgIf.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_NgIf;
}());
